core/src/saros/concurrent/management/JupiterClient.java

```diff
@@ -71,4 +78,26 @@ public synchronized ChecksumActivity withTimestamp(ChecksumActivity checksumActi
 
     return get(checksumActivity.getPath()).withTimestamp(checksumActivity);
   }
+
+  @Override
+  public void start() {
+    jupiterHeartbeat =
+        new ScheduledThreadPoolExecutor(1, new NamedThreadFactory("dpp-jupiter-heartbeat"));
+    jupiterHeartbeat.scheduleWithFixedDelay(this::dispatchHeartbeats, 1, 1, TimeUnit.MINUTES);
+  }
+
+  private synchronized void dispatchHeartbeats() {
+    clientDocs.forEach(
+        (path, jupiter) -> {
+          JupiterActivity heartbeat =
```

Comment:
The weird part here is that you fire a JupiterActivity which then gets passed to some Jupiter component again before it is transmitted to every user.